test-app/app/services/storage.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  type ServiceFactory,
+  provide,
+  service,
+  singleton,
+} from 'ember-polaris-service';
+import Storage from 'ember-storage';
+import { enable } from 'ember-storage/extensions';
+import { type StorageWithBatching } from 'ember-storage-batch';
+
+export class AppStorage extends Storage implements StorageWithBatching {
```

Comment:
This `implements StorageWithBatching` is nice but optional – even if you don't do this, you will still get the type error down below